### PR TITLE
BKNDLSS-13911

### PR DESCRIPTION
**Pull request by olgadanylova**
http://bugs.backendless.com/browse/BKNDLSS-13911

### CODE REVIEW

**Comment by markpiller** [comment]
The branch has conflicts and cannot be merged automatically. Please look into it.

File: SDK/backendlessAPI/Classes/Persistence/DataStoreFactory.m [hunk 1]
```diff
@@ -224,6 +235,18 @@ -(void)findById:(id)objectID response:(void(^)(id))responseBlock error:(void(^)(
             [backendless.persistenceService findByObject:objectID response:responseBlock error:errorBlock];
 }
 
+-(void)findById:(id)objectID queryBuilder:(DataQueryBuilder *)queryBuilder response:(void(^)(id))responseBlock error:(void(^)(Fault *))errorBlock {
+    
+    if ([objectID isKindOfClass:[NSString class]])
```
Comment (ksv510) [inline SDK/backendlessAPI/Classes/Persistence/DataStoreFactory.m]: add curly brackets

File: SDK/backendlessAPI/Classes/Persistence/DataStoreFactory.m [hunk 2]
```diff
@@ -215,24 +226,44 @@ -(void)findLast:(int)relationsDepth response:(void(^)(id result))responseBlock e
...
             [backendless.persistenceService findByObject:objectID response:responseBlock error:errorBlock];
+    }
+}
+
+-(void)findById:(id)objectID queryBuilder:(DataQueryBuilder *)queryBuilder response:(void(^)(id))responseBlock error:(void(^)(Fault *))errorBlock {
+    
+    if ([objectID isKindOfClass:[NSString class]]) {
+        [backendless.persistenceService findByClassId:_entityClass objectID:objectID queryBuilder:queryBuilder response:responseBlock error:errorBlock];
+    }
+
```
Comment (ksv510) [inline SDK/backendlessAPI/Classes/Persistence/DataStoreFactory.m]: redundant new line break

File: SDK/backendlessAPI/Classes/Persistence/DataStoreFactory.m [hunk 3]
```diff
@@ -215,24 +226,44 @@ -(void)findLast:(int)relationsDepth response:(void(^)(id result))responseBlock e
...
 }
 
 -(void)findById:(id)objectID relationsDepth:(int)relationsDepth response:(void(^)(id result))responseBlock error:(void(^)(Fault *))errorBlock {
     
-    if ([objectID isKindOfClass:[NSString class]])
+    if ([objectID isKindOfClass:[NSString class]]) {
         [backendless.persistenceService findById:NSStringFromClass(_entityClass) sid:objectID relations:@[] relationsDepth:relationsDepth response:responseBlock error:errorBlock];
-    else
+    }
+
```
Comment (ksv510) [inline SDK/backendlessAPI/Classes/Persistence/DataStoreFactory.m]: the same

**Pull request merged**